cluster.go

```diff
@@ -664,6 +675,37 @@ func (c *clusterClient) doresultfn(
 			} else {
 				nc = c.redirectOrNew(addr, cc, cm.Slot(), mode)
 			}
+			if hasInit && ei < i { // find out if there is a transaction block or not.
+				for mi = i; mi >= 0 && !isMulti(commands[mi]) && !isExec(commands[mi]); mi-- {
+				}
+				for ei = i; ei < len(commands) && !isMulti(commands[ei]) && !isExec(commands[ei]); ei++ {
+				}
+				if mi >= 0 && mi < ei && ei < len(commands) && isMulti(commands[mi]) && isExec(commands[ei]) && resps[mi].val.string == ok { // a transaction is found.
+					mu.Lock()
+					retries.Redirects++
+					nr := retries.m[nc]
+					if nr == nil {
+						nr = retryp.Get(0, len(commands))
+						retries.m[nc] = nr
+					}
+					for i := mi; i <= ei; i++ {
+						ii := cIndexes[i]
+						cm := commands[i]
+						if mode == RedirectAsk {
+							nr.aIndexes = append(nr.aIndexes, ii)
+							nr.cAskings = append(nr.cAskings, cm)
+						} else {
+							nr.cIndexes = append(nr.cIndexes, ii)
+							nr.commands = append(nr.commands, cm)
+						}
+					}
+					mu.Unlock()
+					continue // the transaction has been added to the retries, go to the next cmd.
+				}
+			}
+			if hasInit && (mi < i && i < ei && mi >= 0 && isMulti(commands[mi]) || i > ei && ei >= 0 && isMulti(commands[ei])) {
```

Comment:
`isMulti(commands[ei])` maybe a typo ? should be `isExec(commands[ei])`

Reply:
It is not a typo. If `commands[ei]` is a `MULTI` then we should not retry the command on `i`. However, I found that it is actually a dead code, `i` will never be larger than `ei` at this point, so the condition is removed now.